docker/README.md

```diff
@@ -3,166 +3,264 @@ Licensed under Creative Commons Attribution 4.0 International License
-cannot re-use DNS config from /etc/resolv.conf as in this case.
-See [this](https://github.com/moby/moby/pull/37485)
-and [this](https://github.com/docker/libnetwork/issues/2068) link for
-background.
-
-# Getting Started
-
-The easiest way to get started is to run
+
+The `start_development.sh` script can be used in the `ccf_container`
+in the same way it is used in the `services_container` above.
+
+Note: be sure to run `make clean` in the host (out of container) build
+tree.
+
+<!--- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx --->
+## Pattern: Service Deployment ##
+
+The containers can be used to deploy network services from the
+`pdo_ccf` and `pdo_services` images. Set the environment variables
+`PDO_HOSTNAME` to the interface where the service will listen and
+`PDO_LEDGER_URL` as the endpoint for CCF.
+
+The scripts used to start CCF, PDO services and clients take a number
+of options that can be passed after the docker image name. While each
+script may have unique parameters (try running with `--help`), there
+are several that are common to all of the start up scripts:
+
+* `--interface` : the default hostname for providing or accessing services
+* `--ledger` : the URL for the ledger
+* `--mode <build|copy|skip>` : the mode for handling
+  configuration; `build` creates a new set of configuration files
+  using the hostname in the interface parameter, `copy` copies the
+  configuration files from the transfer directory, and `skip` does
+  nothing
+
+For example, the following command will start the PDO services using
+`localhost` interface for exporting services:
+
 ```bash
-make test-with-ccf
+    docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer --network host \
+        --name ${USER}_services_container pdo_services \
+        --interface localhost --ledger http://127.0.0.1:6600
 ```
-which will build your current locally committed branch and run the
-end-to-end tests based on a completely freshly setup ccf ledger
-in default configuration.  Starting from a good branch, this should
-also normally successfully run and hence allows also making sure that
-the docker setup is ok.
-
-# End-to-end testing
-
-You are also encouraged to run `make test` whenever you create a new pull
-request as it is easiest way to robustly check your commits. Doing so
-doesn't require any deeper understanding of docker or docker-compose
-and should run out-of the box.
-
-By default it will build and run tests only in SGX simulator mode.
-For testing in SGX hardware mode,
-please see this [section](../docs/install.md#intel-software-guard-extensions-sgx)
-to set up your configuration.
-
-# Developing and manual testing with docker(-compose)
-
-If you want to run your own experiments in the test environment, you
-can run
- ```bash
- mkdir -p $PDO_LEDGER_KEY_ROOT
- make test-env-setup(-ccf)
- cp ccf_keys/*.pem $PDO_LEDGER_KEY_ROOT # only for CCF ledger
+
+** NOTE: We need a better way to process registrations for SGX HW mode. In
```

Comment:
Nit:
```suggestion
:warning: **NOTE: We need a better way to process registrations for SGX HW mode. In
```

Reply:
suggest we just leave it as is.

Reply:
slight modification, only bolding "NOTE"